src/Acuminator/Acuminator.Analyzers/Analyzers/DAC/DacPropertyGraphUsageAnalyzer.cs

```diff
@@ -0,0 +1,135 @@
+﻿using Acuminator.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Acuminator.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class DacPropertyGraphUsageAnalyzer : PXDiagnosticAnalyzer
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+            ImmutableArray.Create
+            (
+                Descriptors.PX1029_PXGraphUsageInDacProperty
+            );
+
+        internal override void AnalyzeCompilation(CompilationStartAnalysisContext compilationStartContext, PXContext pxContext)
+        {
+            compilationStartContext.RegisterSyntaxNodeAction(syntaxContext => AnalyzeGraphUsageInsideDacProperty(syntaxContext, pxContext), SyntaxKind.PropertyDeclaration);
+        }
+
+        private void AnalyzeGraphUsageInsideDacProperty(SyntaxNodeAnalysisContext syntaxContext, PXContext pxContext)
+        {    
+            PropertyDeclarationSyntax node = syntaxContext.Node as PropertyDeclarationSyntax;
+            IPropertySymbol property = syntaxContext.SemanticModel.GetDeclaredSymbol(node, syntaxContext.CancellationToken);
+
+            if (property != null && IsDacProperty(property, pxContext))
+            {
+                IEnumerable<MemberAccessExpressionSyntax> membersAccess = node.DescendantNodes().OfType<MemberAccessExpressionSyntax>();
```

Comment:
@metallmonster, you also need to consider ConditionalMemberAccess nodes, its for syntax like:
`graph?.Method();`

Reply:
@SENya1990 ok

Reply:
@metallmonster in the BQL parameters count mismatch diagnostic there was some similar logic, maybe it will help:
https://github.com/Acumatica/Acuminator/blob/master/src/Acuminator/Acuminator.Analyzers/Analyzers/BQL/BqlParameterMismatch/BqlParameterMismatchAnalyzer.cs